test/unit/Elsa.Activities.UnitTests/Helpers/ActivityTestHelper.cs

```diff
@@ -28,16 +28,33 @@ namespace Elsa.Activities.UnitTests.Helpers;
 public static class ActivityTestHelper
 {
     /// <summary>
-    /// Executes an activity in isolation for unit testing purposes and returns the execution context.
-    /// This method handles all the complexity of setting up the execution context,
-    /// evaluating inputs, and executing the activity.
+    /// Executes an activity for unit testing purposes with comprehensive configuration options.
+    /// This unified method handles all activity execution scenarios. When child activities are provided,
+    /// background execution mode is automatically enabled to capture scheduled activities.
     /// </summary>
     /// <param name="activity">The activity to execute</param>
-    /// <param name="configureServices">An optional action to configure the service collection for dependency injection during activity execution.</param>
+    /// <param name="configureServices">Optional action to configure the service collection for dependency injection</param>
+    /// <param name="childActivities">Optional child activities that should be part of the workflow graph. When provided, automatically enables background execution mode to capture scheduled activities.</param>
     /// <returns>The ActivityExecutionContext used for execution</returns>
-    public static async Task<ActivityExecutionContext> ExecuteActivityAsync(IActivity activity, Action<IServiceCollection>? configureServices = null)
+    public static async Task<ActivityExecutionContext> ExecuteActivityAsync(
+        IActivity activity, 
+        Action<IServiceCollection>? configureServices = null,
+        IEnumerable<IActivity>? childActivities = null)
     {
-        var context = await CreateMinimalActivityExecutionContext(activity, configureServices);
+        ActivityExecutionContext context;
+
+        if (childActivities != null && childActivities.Any())
+        {
+            // Create workflow with child activities for activity scheduling scenarios
+            context = await CreateActivityExecutionContextWithChildActivities(activity, childActivities, configureServices);
+            // Enable background execution mode to capture scheduled activities
+            context.SetIsBackgroundExecution();
```

Comment:
You should be able to capture scheduled activities without specifying this background execution, like this:

```csharp
context.WorkflowExecutionContext.Scheduler.Find(x => ...)
```

Reply:
Yes, because the scheduler is a substitute. If we register the queue based scheduler, it will work (when also removing the SetBackground call).

```csharp
services.AddSingleton<IActivitySchedulerFactory, ActivitySchedulerFactory>();
```